CRM/Financial/BAO/FinancialType.php

```diff
@@ -345,6 +345,34 @@ public static function getAvailableMembershipTypes(&$membershipTypes = NULL, $ac
     return $membershipTypes;
   }
 
+  /**
+   * This function adds the Financial ACL clauses to the where clause.
+   *
+   * This is currently somewhat mocking the native hook implementation
+   * combined with applying the acls in core. This is seen as a transitional step
+   * as it would not need to call the hook if the financialreportsacl
+   * extension is installed. And we probably need to check if it IS
+   * installed before adding ACLs as part of the transition.
+   *
+   * @param array $whereClauses
+   */
+  public static function addACLClausesToWhereClauses(&$whereClauses) {
+    CRM_Utils_Hook::selectWhereClause('Contribution', $whereClauses);
```

Comment:
Won't it make sense to shift this line at 373 if someone wants to alter the financial type clause based on their requirements.?

Reply:
@pradpnayak yes that makes sense - the big challenge is that I'm trying to figure out how to deal with the fact that some of the contribution financial type acls are in core & some are in the extension & if we standardise them more in core people with the extension will get both - my best guess is for now to only call the financial type acl stuff in core if the above call changes nothing - & we can move to phase our of core - do you have any thoughts?

Reply:
@pradpnayak also note that the extension would correctly filter on line items - per comments this gives a wrong answer due to the way line items are added

Reply:
Yes you are right,[ Financial Type ACL report ](https://github.com/JMAConsulting/biz.jmaconsulting.financialaclreport/blob/master/financialaclreport.php#L107) extension also uses selectWhereClause() hook. 

> my best guess is for now to only call the financial type acl stuff in core if the above call changes nothing - & we can move to phase our of core 

+1